packages/dynamic-links/android/src/main/java/io/invertase/firebase/dynamiclinks/ReactNativeFirebaseDynamicLinksModule.java

```diff
@@ -106,25 +125,38 @@ public void buildShortLink(ReadableMap dynamicLinkMap, String shortLinkType, Pro
 
   @ReactMethod
   public void getInitialLink(Promise promise) {
+    // Check if getDynamicLink() has already completed successfully.
+    // This ensures the initial link is returned once, if found.
     if (gotInitialLink) {
-      if (initialLinkUrl != null) {
-        promise.resolve(dynamicLinkToWritableMap(initialLinkUrl, initialLinkMinimumVersion));
-      } else {
-        promise.resolve(null);
-      }
+      promise.resolve(null);
+      return;
+    }
 
+    // Check for the case where getInitialLink
+    // runs before the LifeCycleState is RESUMED (e.g. BEFORE_CREATE or BEFORE_RESUME).
+    if(!hostResumed) {
+      // Use initialPromise to store the Promise that was passed and
+      // run it when LifeCycleState changes to RESUMED in onHostResume.
+      initialPromise = promise;
       return;
     }
 
     Activity currentActivity = getCurrentActivity();
+    // Shouldn't happen anymore, left as a guard.
     if (currentActivity == null) {
       promise.resolve(null);
       return;
     }
 
-    FirebaseDynamicLinks.getInstance().getDynamicLink(currentActivity.getIntent())
+    Intent currentIntent = currentActivity.getIntent();
+    // Verify if the app was resumed from the Overview (history) screen.
+    launchedFromHistory = currentIntent != null && (currentIntent.getFlags() & Intent.FLAG_ACTIVITY_LAUNCHED_FROM_HISTORY) != 0;
```

Comment:
```suggestion
    launchedFromHistory = (currentIntent != null) && ((currentIntent.getFlags() & Intent.FLAG_ACTIVITY_LAUNCHED_FROM_HISTORY) != 0);
```

This is pedantic, but I don't like forcing developers to think through precedence of operators, so I prefer explicit parentheses at all times